lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2815,62 +2839,76 @@ void PartitionCloner::handleSwitchEnum(SwitchEnumInst *inst) {
-    createConstTensorAttrsOnAccel(caseZero, loc, ctx, BA, constOpName,
-                                  operands);
-    FP.configuration.handleDevicePlacement(
-        "Const", /*opDevice*/ getDeviceString(DeviceType::ALL), BA, loc,
-        operands, constOpName);
-    auto subMap = getSingleSubstitutionMapForElementType(
-        intFieldSILType.getASTType(), ctx);
-    auto tensorHandleInt64Ty =
-        convertElementTypeToTensorValueType(intFieldSILType);
-    zeroConstTensor = BA.createBuiltin(loc, ctx.getIdentifier(constOpName),
-                                       tensorHandleInt64Ty, subMap, operands);
-  }
+  // Let K be the total # cases (including the optional default case). Create a
+  // chain of K-1 BB's with cond_br, to dispatch to the K cases.
+  for (unsigned i = 0, e = caseIt.getNumCases() - 1; i != e; ++i) {
+    // Create a scalar const tensor i, to compare with case id.
+    BuiltinInst *constTensorWithCaseId = nullptr;
+    {
+      auto *caseIdIntInst = BA.createIntegerLiteral(loc, intFieldSILType, i);
+      SmallVector<SILValue, 4> operands;
+      std::string constOpName = "__tfop_Const";
+      createConstTensorAttrsOnAccel(caseIdIntInst, loc, ctx, BA, constOpName,
+                                    operands);
+      FP.configuration.handleDevicePlacement(
+          "Const", /*opDevice*/ getDeviceString(DeviceType::ALL), BA, loc,
+          operands, constOpName);
+      auto subMap = getSingleSubstitutionMapForElementType(
+          intFieldSILType.getASTType(), ctx);
+      auto tensorHandleInt64Ty =
+          convertElementTypeToTensorValueType(intFieldSILType);
+      constTensorWithCaseId =
+          BA.createBuiltin(loc, ctx.getIdentifier(constOpName),
+                           tensorHandleInt64Ty, subMap, operands);
+    }
 
-  // Cast to a TensorHandle<Builtin.Int1> value
-  BuiltinInst *condBrOperand = nullptr;
-  {
-    // Omit the metatype attr T for simplicity, and TF graphDef compiler can
-    // infer the type.
-    std::string equalOpName = "__tfop_Equal,$in,$in";
-    auto boolFieldSILType =
-        extractBuiltinTypeFromStdlibNumericType(ctx.getBoolDecl());
-    auto tensorHandleI1Ty =
-        convertElementTypeToTensorValueType(boolFieldSILType);
-    SmallVector<SILValue, 4> operands = {acceleratorCaseId, zeroConstTensor};
-    FP.configuration.handleDevicePlacement(
-        "Equal", /*opDevice*/ getDeviceString(DeviceType::ALL), BA, loc,
-        operands, equalOpName);
-    auto subMap = getSingleSubstitutionMapForElementType(
-        boolFieldSILType.getASTType(), ctx);
-    auto *equalComparisonInst =
-        BA.createBuiltin(loc, ctx.getIdentifier(equalOpName), tensorHandleI1Ty,
-                         subMap, operands);
-    condBrOperand =
-        BA.createBuiltin(loc, ctx.getIdentifier("tf_tensor_to_i1"),
-                         boolFieldSILType, {}, {equalComparisonInst});
-  }
+    // Cast to a TensorHandle<Builtin.Int1> value
```

Comment:
Instead of 'cast', maybe "compare it to a constant to get a Tensorhandle<Builtin.Int1> value"?

Reply:
Fixed